feature-libs/cart/quick-order/core/services/quick-order.service.spec.ts

```diff
@@ -325,24 +325,20 @@ describe('QuickOrderService', () => {
       });
   });
 
-  it('should add deleted entry and after 7s delete it', (done) => {
+  it('should add deleted entry and after 7s delete it', fakeAsync(() => {
     service.loadEntries(mockEntries);
     service.softDeleteEntry(0);
+    let result: Record<string, OrderEntry>;
 
-    service
-      .getSoftDeletedEntries()
-      .pipe(
-        tap((softDeletedEntries) => {
-          expect(softDeletedEntries).toEqual({ mockCode1: mockEntry1 });
-        }),
-        delay(7000),
-        switchMap(() => service.getSoftDeletedEntries())
-      )
-      .subscribe((result) => {
-        expect(result).toEqual({});
-      });
-    done();
-  });
+    service.getSoftDeletedEntries().subscribe((val) => {
+      result = val;
+    });
+    expect(result).toEqual({ mockCode1: mockEntry1 });
+
+    tick(7000);
+
+    expect(result).toEqual({});
+  }));
```

Comment:
[NITPICK]
For memory safety, let's unsubscribe, when we no longer need the subscription.